modin/experimental/core/storage_formats/hdk/query_compiler.py

```diff
@@ -557,18 +557,34 @@ def concat(self, axis, other, **kwargs):
 
     def drop(self, index=None, columns=None, errors: str = "raise"):
         if index is not None:
-            raise NotImplementedError("Row drop")
+            # Only column drop is supported by the HDK engine
+            raise NotImplementedError("Index drop")
         if errors != "raise":
             raise NotImplementedError(
                 "This lazy query compiler will always "
                 + "raise an error on invalid columns."
             )
-        return self.__constructor__(
-            self._modin_frame.take_2d_labels_or_positional(
-                row_labels=index, col_labels=self.columns.drop(columns)
-            )
+
+        columns = self.columns.drop(columns)
+        new_frame = self._modin_frame.take_2d_labels_or_positional(
+            row_labels=index, col_labels=columns
         )
 
+        # If all columns are dropped and the index is trivial, we are
+        # not able to restore it, since we don't know the number of rows.
+        # In this case, we calculate the index from the current frame.
+        if len(columns) == 0 and new_frame._index_cache is None:
+            if self._modin_frame._index_cache is not None:
+                new_frame._index_cache = self._modin_frame._index_cache
+            else:
+                # Lazy execution must be disabled before the index calculation
+                mode = self._modin_frame.force_execution_mode
```

Comment:
I don't think it is a good idea to reset the execution mode here. It was introduced for testing purposes to have the ability to make sure we have a lazy execution in a test. If we cannot execute some test lazily, then it just shouldn't try to force lazy execution.

Reply:
Removed the execution mode reset.